serveradmin/servershell/views.py

```diff
@@ -156,12 +198,17 @@ def get_results(request):
         restrict = shown_attributes.copy()
         if 'servertype' not in restrict:
             restrict.append('servertype')
-        query = Query(parse_query(term), restrict, order_by)
+        main_query = Query(parse_query(term), restrict, order_by)
+
+        query = MergedQuery([
+            Query({'object_id': Any(*pinned)}, restrict, None),
+            main_query,
+        ])
 
         # TODO: Using len is terribly slow for large datasets because it has
         #  to query all objects but we cannot use count which is available on
         #  Django QuerySet
-        num_servers = len(query)
+        num_servers = len(list(query))
```

Comment:
Wouldn't this give us a list of Queries ?

Reply:
`list()` calls the `__iter__()` method on the `MergedQuery` so it gets a `MergedQueryIterator` and iterates over it to create a list out of its items.